script.js

```diff
@@ -0,0 +1,315 @@
+          key.innerText = keysUnshift[index];
+        }
+        if(key.innerText.length == 1){
+          if(key.classList.contains('uppercase') ){
+            key.classList.remove('uppercase');
+            key.classList.add('lowercase');
+          }else{
+            key.classList.remove('lowercase');
+            key.classList.add('uppercase');
+          }
+        }
+      })
+    }
+  }
+
+  // Changes the keyboard's language (eng or rus)
+    //@e{object} -  addEventListener function event
+    //return:  @lang{string}: eng/rus - current keyboard language
+
+  function changeLang(e){
+    if(e.ctrlKey && e.altKey){
+      if(keyboard.classList.contains('eng')){
+        keyboard.classList.add('rus');
+        keyboard.classList.remove('eng');
+        Array.from(keyboard.children).forEach((key, index) =>{
+          key.innerText = keysRus[index];
+        })
+      }else{
+        keyboard.classList.add('eng');
+        keyboard.classList.remove('rus');
+        Array.from(keyboard.children).forEach((key, index) =>{
+          key.innerText = keysEng[index];
+        })
+      }
+      setLastLang();
+    }
+    let lang;
+    if (keyboard.classList.contains('eng')){
+      lang = 'eng'
+      shiftKeyPress(e, keysEng, keysEngShift);
+    }else{
+      lang = 'rus';
+      shiftKeyPress(e, keysRus, keysRusShift);
+    }
+    return lang;
+  }
+
+  // Add 'active' class to the letters in css
+
+  function keyActive(e){
+    let index = keyCodes.indexOf(e.code);
+    if(e.type == 'keydown' && index != -1){
+      keyboard.children[index].classList.add('key_active');
+    }
+    if(e.type == 'keyup' && index != -1){
+      keyboard.children[index].classList.remove('key_active');
+    }
+    if(e.target.classList.contains('key')){
+      if(e.type == 'mousedown'){
+        e.target.classList.add('key_active');
+      }
+      if(e.type == 'mouseup'){
+        e.target.classList.remove('key_active');
+      }
+    }
+  }
+
+  // Letters input with mouse
+
+  function mouseEvent(e) {
+    keyActive(e);
+    changeLang(e);
+    if(e.type == 'mousedown'){
+      capsLockKeyPress(e);
+      let key = e.target.innerText;
+      if(e.target.classList.contains('key')){
+        if(key === ''){
+          entryField.value += ' ';
+        }else if(key == 'Backspace'){
+          entryField.value = entryField.value.substring(0, entryField.value.length - 1);
+        }else if(key == 'Tab'){
+          entryField.value += '   ';
+        }else if(key == 'Enter'){
+          entryField.value += '\n';
+        }else if(key == 'Alt'
+              || key == 'Ctrl'
+              || key == 'Win'
+              || key == 'CapsLock'
+              || key == 'Meta'
+              || key == 'Shift'
+              || key == 'Del'){
+          entryField.value += '';
+        }else{
+          entryField.value += key;
+        }
+      }
+    }
+  }
+
+  // Letters input from the physical keyboard
+
+  function keyEvent(e){
```

Comment:
Too much complexity for one function. Can be splitted for several functions.

Reply:
`function keyEvent(e)`  now contains new simply functions:
-  inputSpecialKeys(e);
- letterCase(e, index, keysUnshift, keysShift) - i know it's a bad name, but I couldn't think of another